pulsar-client/src/main/java/org/apache/pulsar/client/impl/conf/ClientConfigurationData.java

```diff
@@ -377,6 +380,29 @@ public Authentication getAuthentication() {
     public void setAuthentication(Authentication authentication) {
         this.authentication = authentication;
     }
+
+    /**
+     * Set the authentication using auth-related configuration properties if they are available.
+     * <p>
+     * The {@code authPluginClassName} and one of {@code authParams} or {@code authParamMap} must be configured for
+     * the authentication to be determined and set.
+     *
+     * @see AuthenticationFactory#create
+     * @throws PulsarClientException if authentication creation fails
+     */
+    public void setAuthenticationFromPropsIfAvailable() throws PulsarClientException {
```

Comment:
Naming... 🤷🏻

Reply:
It looks like we move this to `getAuthentication()`.